multicluster/controllers/multicluster/stale_controller.go

```diff
@@ -270,6 +295,54 @@ func (c *StaleResCleanupController) cleanupClusterInfoResourceExport(commonArea
 	return nil
 }
 
+func (c *StaleResCleanupController) cleanupMemberClusterAnnounce() error {
+	clusterSetList := &mcsv1alpha1.ClusterSetList{}
+	if err := c.List(ctx, clusterSetList, &client.ListOptions{}); err != nil {
+		return err
+	}
+	memberClusterAnnounceList := &mcsv1alpha1.MemberClusterAnnounceList{}
+	if err := c.List(ctx, memberClusterAnnounceList, &client.ListOptions{}); err != nil {
+		return err
+	}
+
+	var members []mcsv1alpha1.MemberCluster
+	if len(clusterSetList.Items) != 0 {
```

Comment:
Probably just check `if len(clusterSetList.Items) == 1` and add a comment to indicates that ClusterSet webhook will guarantee the maximum number of clusterSet will be 1.